app/src/main/java/net/squanchy/navigation/view/BottomTabsPagesFactory.java

```diff
@@ -0,0 +1,23 @@
+package net.squanchy.navigation.view;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import net.squanchy.R;
+
+class BottomTabsPagesFactory {
+
+    static View inflate(Context context, ViewGroup parent, @Tab int position){
+        LayoutInflater inflater = LayoutInflater.from(context);
+        switch (position){
+            case BottomNavigationPagerAdapter.SCHEDULE_POSITION:
+                return inflater.inflate(R.layout.schedule_view, parent, false);
+            case BottomNavigationPagerAdapter.FAVOURITES_POSITION:
+            case BottomNavigationPagerAdapter.TWEETS_POSITION:
+            case BottomNavigationPagerAdapter.VENUE_POSITION:
+        }
+        throw new IllegalArgumentException("Invalid position");
```

Comment:
Why outside of the `switch`? It should be in the `default` branch (which in this case you can leave collapsed with the unhandled cases). Also maybe a more informative message such as `Unsupported tab type for position n` could help